snow/networking/handler/handler.go

```diff
@@ -556,23 +557,11 @@ func (h *handler) handleSyncMsg(ctx context.Context, msg Message) error {
 		return engine.GetStateSummaryFrontierFailed(ctx, nodeID, msg.RequestID)
 
 	case *p2p.GetAcceptedStateSummary:
-		// TODO: Enforce that the numbers are sorted to make this verification
-		//       more efficient.
-		if !utils.IsUnique(msg.Heights) {
-			h.ctx.Log.Debug("message with invalid field",
-				zap.Stringer("nodeID", nodeID),
-				zap.Stringer("messageOp", message.GetAcceptedStateSummaryOp),
-				zap.Uint32("requestID", msg.RequestId),
-				zap.String("field", "Heights"),
-			)
-			return engine.GetAcceptedStateSummaryFailed(ctx, nodeID, msg.RequestId)
-		}
-
```

Comment:
I guess we're fine accepting malformed responses, now that we're sanitizing them? I think it'd be reasonable to keep this snippet, but not necessary

Reply:
I want to pass in a `set.Set` for all of the types that should be unique... We could still enforce that we only handle unique elements... But I don't think we'd ever want to use `IsUnique` - because we should just be using the provided set.

If we wanted to keep the uniqueness - we should verify that the length of the set is equal to the length of the initially provided slice.

Reply:
talked offline - going to remove the check